app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.java

```diff
@@ -524,6 +537,18 @@ public void markAllAsRead(int position){
                 unreadCount.setVisibility(count == 0 ? View.INVISIBLE : View.VISIBLE);
                 Toast.makeText(getActivity(), "Marked all as read", Toast.LENGTH_SHORT).show();
             }
+
+            public void markFavoriteStatus(int position, boolean isFavourite){
+                Course course = mCourseList.get(position);
+                courseDataHandler.markFavoriteStatus(course.getCourseId(), isFavourite);
+                course.setFavoriteStatus(isFavourite);
+                mCourseList = courseDataHandler.rearrangeCourses(mCourseList);
+                notifyDataSetChanged();
+
+                favorite.setVisibility(isFavourite ? View.VISIBLE : View.INVISIBLE);
```

Comment:
If I'm not wrong, calling `notifyDataSetChanged()` will cause the LayoutManager to rebind all views. This function call wouldn't be necessary in that case.